### ISSUE TITLE
Provide RFC9557 patterns

### ISSUE THREAD

**Issue by cromefire** [nodatime/nodatime#1799]
Currently NodaTime [provides ISO and RFC3339 patterns](https://github.com/nodatime/nodatime/blob/8c41fe377b8ff8121b2942e749c2a3b81e797e20/src/NodaTime/Text/ZonedDateTimePattern.cs#L52), but now [RFC9557](https://www.rfc-editor.org/rfc/rfc9557.html) is ratified and [will also be used in JavaScripts upcoming Temporal API](https://github.com/tc39/proposal-temporal/issues/1450).

So it would be great if NodaTime would provide both a pattern to provide it to it's users in an easy way (including the JSON options like [already present for ISO](https://github.com/nodatime/nodatime.serialization/blob/837968481e5cfd9cea42298e17fdc3138877ec66/src/NodaTime.Serialization.SystemTextJson/Extensions.cs#L54), although that probably needs a follow-up issue on that repo) and also (and here I haven't tested yet if it already works, but I don't think it does) be able to parse these dates out of the box.

RFC9557 is an extension of RFC3339, so it should mostly affect the ZonedDateTime, everything else should probably stay like it was in RFC3339.

**Comment by cromefire** [comment]
I think the most important things would be the time zone and the calendar, at least for my use case those 2 would be the most important metadata.

**Comment by jskeet** [comment]
Right, but we'd need to implement the overall parsing to handle extended information. If this is urgent for you, you're welcome to work on it yourself - but if it's left to me I think it's unlikely that I'd have anything ready to release until at least the end of June.

**Comment by cromefire** [comment]
I'm mostly interested in the serialization for now so maybe I'll give that a shot some time, but it's not that urgent for me either. Currently in the one project where I use Temporal (with a polyfill) and NodaTime, I just format it manually.

It has just been a standard for a few days now, but in the long run it'll hopefully be as ubiquitous as RFC3339 and serve as a common ground for the use of zoned date times, especially with web APIs, especially once Temporal becomes the standard for JS-based stuff.

**Comment by jskeet** [comment]
One issue with the calendar aspect is making sure everyone agrees on *exactly* what calendar we're talking about for any given identifier. I guess that's a CLDR issue more than anything else... I should probably add a CLDR ID property to CalendarSystem (and a reverse mapping).

**Comment by cbodin** [comment]
For anyone else ending up in this thread, a custom format can be used to serialize/deserialize to RFC9557 (not including calendar id):
```csharp
var pattern = ZonedDateTimePattern.CreateWithInvariantCulture(
        "uuuu'-'MM'-'dd'T'HH':'mm':'sso<+HH:mm>'['z']'",
        DateTimeZoneProviders.Tzdb);

// Deserialize
var result = pattern.Parse("2025-03-31T13:00:00+02:00[Europe/Stockholm]");

// Serialize
var text = pattern.Format(result.Value); // 2025-03-31T13:00:00+02:00[Europe/Stockholm]
```
